backoff/retry.go

```diff
@@ -0,0 +1,41 @@
+package backoff
+
+import (
+	"time"
+
+	"github.com/utilitywarehouse/semaphore-wireguard/log"
+)
+
+type Operation func() error
+
+const (
+	defaultBackoffJitter = true
+	defaultBackoffMin    = 2 * time.Second
+	defaultBackoffMax    = 1 * time.Minute
+)
+
+func RetryWithDefaultBackoff(op Operation, description string) {
```

Comment:
```suggestion
func Retry(op Operation, description string) {
```
Nitpick. I don't think you need to be explicit about the fact that it uses defaults.

Reply:
hmm I like the explicit names, but I do not have a strong argument. In that case I guess we should only care to export `Retry` from the package and hide everything else.

Reply:
I think exporting `backoff.Retry` and `backoff.RetryWithBackoff` is fine.

My thinking is that, i.e `http.Get` doesn't need to be called `http.GetWithDefaultClientAndSettings` for me to understand that it's going to use some defaults.